### PR TITLE
Added extensions on IServiceCollection

### LINKED ISSUE #66: Add IServiceCollection extension methods to enable serilog
With https://github.com/dotnet/runtime/issues/61634#issuecomment-1033038101 and https://github.com/dotnet/runtime/pull/65109, Microsoft.Extensions.Hosting added a new way of creating a hosted app that doesn't use IHostBuilder.

We should add overloads to `SerilogHostBuilderExtensions` that accept an `IServiceCollection` instead of an `IHostBuilder`. This will allow consumers using the new HostApplicationBuilder to use Serilog easily.

Since the methods will extend `IServiceCollection`, the methods should probably be named `AddSerilog` instead of `UseSerilog`. This follows the naming patterns used elsewhere.

See also https://github.com/dotnet/runtime/pull/68580 and https://github.com/dotnet/aspnetcore/pull/43056

cc @davidfowl

### PR DESCRIPTION
**Pull request by mburumaxwell**
This adds new extensions on `IServiceCollection` similar to those on `IHostBuilder`. Also changed the implementation of the ones in `IHostBuilder` to rely on the new extensions for easier maintenance in the future. The implementation logic as pertains to registration of services remains the same.

Fixes: #66, #68

### CODE REVIEW

File: src/Serilog.Extensions.Hosting/SerilogServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Serilog.Extensions.Hosting;
+using Serilog.Extensions.Logging;
+// ReSharper disable MemberCanBePrivate.Global
+
+namespace Serilog
+{
+    /// <summary>
+    /// Extends <see cref="IHostBuilder"/> with Serilog configuration methods.
```
Comment (sungam3r) [inline src/Serilog.Extensions.Hosting/SerilogServiceCollectionExtensions.cs]:
```suggestion
    /// Extends <see cref="IServiceCollection"/> with Serilog configuration methods.
```

File: test/Serilog.Extensions.Hosting.Tests/SerilogServiceCollectionExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        public void ServicesAreRegisteredWhenCallingAddSerilog()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+
+            // Act
+            collection.AddSerilog();
+
+            // Assert
+            IServiceProvider provider = collection.BuildServiceProvider();
```
Comment (sungam3r) [inline test/Serilog.Extensions.Hosting.Tests/SerilogServiceCollectionExtensionsTests.cs]:
```suggestion
            using var provider = collection.BuildServiceProvider();
```

File: test/Serilog.Extensions.Hosting.Tests/SerilogServiceCollectionExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            ILogger logger = new LoggerConfiguration().CreateLogger();
+
+            // Act
+            collection.AddSerilog(logger);
+
+            // Assert
+            IServiceProvider provider = collection.BuildServiceProvider();
```
Comment (sungam3r) [inline test/Serilog.Extensions.Hosting.Tests/SerilogServiceCollectionExtensionsTests.cs]:
```suggestion
            using var provider = collection.BuildServiceProvider();
```

File: test/Serilog.Extensions.Hosting.Tests/SerilogServiceCollectionExtensionsTests.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        public void ServicesAreRegisteredWhenCallingAddSerilogWithConfigureDelegate()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+
+            // Act
+            collection.AddSerilog(_ => { });
+
+            // Assert
+            IServiceProvider provider = collection.BuildServiceProvider();
```
Comment (sungam3r) [inline test/Serilog.Extensions.Hosting.Tests/SerilogServiceCollectionExtensionsTests.cs]:
```suggestion
            using var provider = collection.BuildServiceProvider();
```

**Comment by nblumhardt** [comment]
LGTM! Thanks @mburumaxwell and @sungam3r for the review 👍 

I'll bump the dev package version to 5.1.0-* to indicate new feature additions.

**Comment by AraHaan** [comment]
Can a followup pr be made to split the servicecollection versions over into an ``Serilog.Extensions.DependencyInjection`` library so that way those who do not want to use the hosting packages could get the benefits of using the new extensions?

Also, then the hosting package could depend on that package as well which should be ok as Microsoft.Extensions.DependencyInjection would be provided anyways.

**Comment by mburumaxwell** [comment]
> LGTM! Thanks @mburumaxwell and @sungam3r for the review 👍 
> 
> 
> 
> I'll bump the dev package version to 5.1.0-* to indicate new feature additions.

Cool. Looking forward to the release.

**Comment by mburumaxwell** [comment]
> Can a followup pr be made to split the servicecollection versions over into an ``Serilog.Extensions.DependencyInjection`` library so that way those who do not want to use the hosting packages could get the benefits of using the new extensions?
> 
> 
> 
> Also, then the hosting package could depend on that package as well which should be ok as Microsoft.Extensions.DependencyInjection would be provided anyways.

This makes sense. Maybe you can do the PR?

If we have to move things, instead of a new package, I'd move the extensions to `Serilog.Extensions.Logging` (lives in a different repository) which is referenced by the hosting one here. In such a case, I wonder if the hosting package would still be needed seeing that one can do it. Maybe that's a different discussion?

**Comment by AraHaan** [comment]
Yeah, I think putting it in the logging package would be better as well. And yes, I think the hosting package can be deprecated after this (which would help you guys more with less packages to maintain).

**Comment by mburumaxwell** [comment]
@AraHaan I believe that's for @nblumhardt to decide, being the maintainer.

**Comment by AraHaan** [comment]
Yeah, I know that. I was just saying what I was thinking as well.

**Pull request merged**